src/OrchardCore.Modules/OrchardCore.Workflows/Controllers/WorkflowController.cs

```diff
@@ -0,0 +1,51 @@
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using OrchardCore.Workflows.Services;
+
+namespace OrchardCore.Workflows.Controllers
+{
+    public class WorkflowController : Controller
+    {
+        private readonly IWorkflowManager _workflowManager;
+        private readonly IWorkflowDefinitionRepository _workflowDefinitionRepository;
+        private readonly IWorkflowInstanceRepository _workflowInstanceRepository;
+        private readonly ILogger<WorkflowController> _logger;
+
+        public WorkflowController(
+            IWorkflowManager workflowManager,
+            IWorkflowDefinitionRepository workflowDefinitionRepository,
+            IWorkflowInstanceRepository workflowInstanceRepository,
+            ILogger<WorkflowController> logger)
+        {
+            _workflowManager = workflowManager;
+            _workflowDefinitionRepository = workflowDefinitionRepository;
+            _workflowInstanceRepository = workflowInstanceRepository;
+            _logger = logger;
+        }
+
+        public async Task<IActionResult> Start(int workflowDefinitionId, int activityId)
```

Comment:
Post ?

Reply:
Permission, authentication?

Reply:
> Post ?

I considered that, but I wanted to support scenarios where the user can click on a link from an email message to resume a workflow. But perhaps it's not a good idea to allow for that in the first place. Thoughts?

Reply:
> Permission, authentication?

Will add 👍

Reply:
Done and done.

Reply:
Adding a permission check is good, but not sufficient. You should either use [POST] or require a nonce. You can refactor this to have two actions, calling the same common method once they have verified the caller. When you use a POST you don't ignore the AFT,  and you check for permission. When you accept a POST, you require a nonce in the query string and take it as a valid permission. The user is still authenticated but the nonce provides the authorization part.

Reply:
Good points. But if GET requests require a shared access security token, then that eliminates the potential use case where a user simply wants to respond to GET requests. I am wondering why we would not allow anonymous GET requests to be handled by a workflow, while allowing this for controllers. Whether a request is handled by a middle-ware component, a controller or a workflow is an implementation detail, and all of these implementations should allow the author to control whether the resource should be secured or not. Regarding AFT for POST requests; how would a client, e.g. Postman, GitHub etc. generate the AFT?

I'm thinking, what if we allow the user to control whether a workflow requires protection, including the type of such protection? One mechanism could be a SAS token, but maybe we should also allow the user to provide a more static key, like a shared secret. This would be useful for example to implement a workflow that handles GitHub web-hook events. When configuring the web-hook in GitHub, the user can specify this shared secret. And perhaps the user should be able to specify roles that can trigger a workflow, like content permissions.

In any case, I think workflow execution security should be configurable instead of always requiring a token. Handling webhook events is just one scenario, but I'm sure there are others.